### ISSUE TITLE
remote_endpoint: Transport endpoint is not connected

### ISSUE THREAD

**Issue by minaevmike** [nghttp2/nghttp2#526]
I have a simple example 
```c++
   boost::system::error_code ec;    
    nghttp2::asio_http2::server::http2 server;
    boost::asio::ssl::context sslContext(boost::asio::ssl::context::sslv23);
    sslContext.use_private_key_file("/root/prog/AVScan/conf/ssl/avscand.key", boost::asio::ssl::context::pem);
    sslContext.use_certificate_chain_file("/root/prog/AVScan/conf/ssl/avscand.crt");
    nghttp2::asio_http2::server::configure_tls_context_easy(ec, sslContext);
    server.handle("/", [=](const nghttp2::asio_http2::server::request &req, const  nghttp2::asio_http2::server::response &res)
    {
        res.write_head(200);
        res.end("sadsadasd");
    });
    try
    {
        if (server.listen_and_serve(ec, sslContext, "0.0.0.0", "2121"))
        {
        }
    } catch (const std::exception& e)
    {
       std::cout << e.what() << std::endl;
    }
```
when i try to go to this page via chrome i have an exeption `remote_endpoint: Transport endpoint is not connected` but when i set `server.num_threads(12)` it start working fine. So what i am doing wrong?

**Comment by tatsuhiro-t** [comment]
Thank you for reporting this.  The bug was fixed in 2d6211c

**Comment by tatsuhiro-t** [comment]
Closing since fixed version has been released.
